packages/versions/src/lib/checkFuelCoreVersionCompatibility.test.ts

```diff
@@ -0,0 +1,79 @@
+import { checkFuelCoreVersionCompatibility } from './checkFuelCoreVersionCompatibility';
+import * as getSupportedVersionsMod from './getSupportedVersions';
+
+describe('getDifferenceToUserFuelCoreVersion', () => {
+  const v = '0.1.2';
+  const spy = jest.spyOn(getSupportedVersionsMod, 'getSupportedVersions');
+  spy.mockImplementation(() => ({ FUELS: v, FORC: v, FUEL_CORE: v }));
+
+  it.each([
+    {
+      version: '1.1.2',
+      isSupportedResult: {
+        major: false,
+        minor: true,
+        patch: true,
+      },
+    },
+    {
+      version: '1.2.2',
+      isSupportedResult: {
+        major: false,
+        minor: false,
+        patch: true,
+      },
+    },
+    {
+      version: '1.1.3',
+      isSupportedResult: {
+        major: false,
+        minor: true,
+        patch: false,
+      },
+    },
+    {
+      version: '0.2.2',
+      isSupportedResult: {
+        major: true,
+        minor: false,
+        patch: true,
+      },
+    },
+    {
+      version: '0.2.3',
+      isSupportedResult: {
+        major: true,
+        minor: false,
+        patch: false,
+      },
+    },
+    {
+      version: '0.1.3',
+      isSupportedResult: {
+        major: true,
+        minor: true,
+        patch: false,
+      },
+    },
+    {
+      version: '0.1.2',
+      isSupportedResult: {
+        major: true,
+        minor: true,
+        patch: true,
+      },
+    },
+  ])(
+    `Succeeds when checking $version and ${v} equality`,
+    ({ version, isSupportedResult: { major, minor, patch } }) => {
+      const { isMajorSupported, isMinorSupported, isPatchSupported, userVersion } =
+        checkFuelCoreVersionCompatibility(version);
+
+      expect(isMajorSupported).toEqual(major);
+      expect(isMinorSupported).toEqual(minor);
+      expect(isPatchSupported).toEqual(patch);
+
+      expect(userVersion).toEqual(v);
+    }
+  );
```

Comment:
This sort of `it.each([], ...)` abstraction in tests is sketchy and will backfire most of the time.

It might falsely give the impression of a DRY or more advanced code, but it is harder to read and becomes a pain once something fails. Line numbers will be lost, and the only way to locate the problem will be by string-matching things _visually_, because all assertions are fixed and point to the same place.

Additionally, I added an `afterAll` hook to ensure no mocked methods are left behind, plus minor touchups on nomenclatures, and finally removed the same version being mocked on `FUELS` and `FORC` properties to remove any ambiguities since these two are not part of this topic.


```suggestion
import { checkFuelCoreVersionCompatibility } from './checkFuelCoreVersionCompatibility';
import * as getSupportedVersionsMod from './getSupportedVersions';

describe('getDifferenceToUserFuelCoreVersion', () => {
  afterAll(() => jest.resetAllMocks());

  it('should validate all possible version mismatches', () => {
    const supportedVersion = '0.1.2';

    jest.spyOn(getSupportedVersionsMod, 'getSupportedVersions').mockImplementation(() => ({
      FUELS: '1', // not under test
      FORC: '1', // not under test
      FUEL_CORE: supportedVersion,
    }));

    expect(checkFuelCoreVersionCompatibility('1.1.2')).toEqual({
      isMajorSupported: false,
      isMinorSupported: true,
      isPatchSupported: true,
      supportedVersion,
    });

    expect(checkFuelCoreVersionCompatibility('1.2.2')).toEqual({
      isMajorSupported: false,
      isMinorSupported: false,
      isPatchSupported: true,
      supportedVersion,
    });

    expect(checkFuelCoreVersionCompatibility('1.1.3')).toEqual({
      isMajorSupported: false,
      isMinorSupported: true,
      isPatchSupported: false,
      supportedVersion,
    });

    expect(checkFuelCoreVersionCompatibility('0.2.2')).toEqual({
      isMajorSupported: true,
      isMinorSupported: false,
      isPatchSupported: true,
      supportedVersion,
    });

    expect(checkFuelCoreVersionCompatibility('0.2.3')).toEqual({
      isMajorSupported: true,
      isMinorSupported: false,
      isPatchSupported: false,
      supportedVersion,
    });

    expect(checkFuelCoreVersionCompatibility('0.1.3')).toEqual({
      isMajorSupported: true,
      isMinorSupported: true,
      isPatchSupported: false,
      supportedVersion,
    });

    expect(checkFuelCoreVersionCompatibility('0.1.2')).toEqual({
      isMajorSupported: true,
      isMinorSupported: true,
      isPatchSupported: true,
      supportedVersion,
    });
  });
});
```